packages/x-charts/src/BarChart/useBarChartProps.ts

```diff
@@ -95,6 +95,29 @@ export const useBarChartProps = (props: BarChartProps) => {
       })),
     [hasHorizontalSeries, series],
   );
+
+  const defaultXAxis = hasHorizontalSeries ? undefined : defaultBandXAxis;
+  const processedXAxis = React.useMemo(() => {
+    if (!xAxis) {
+      return defaultXAxis;
+    }
+
+    return hasHorizontalSeries
+      ? xAxis
+      : xAxis.map((axis) => ({ scaleType: 'band' as const, ...axis }));
+  }, [defaultXAxis, hasHorizontalSeries, xAxis]);
+
+  const defaultYAxis = hasHorizontalSeries ? defaultBandYAxis : undefined;
+  const processedYAxis = React.useMemo(() => {
+    if (!yAxis) {
+      return defaultYAxis;
+    }
+
+    return hasHorizontalSeries
+      ? yAxis.map((axis) => ({ scaleType: 'band' as const, ...axis }))
+      : yAxis;
+  }, [defaultYAxis, hasHorizontalSeries, yAxis]);
```

Comment:
Should we eventually move this "chart initialization" logic to the `seriesConfig` so we have a unified place/way of doing this?

Not for this PR though. 😆

Reply:
I'm not sure. This logic is specific to the bar chart. How would moving this to a centralized place help? Wouldn't we need to then add information about which kind of chart we're processing series for so we can know how to create the defaults? Not sure if that would help a lot 🤔

Reply:
`seriesConfig` is specific to each chart. Name is a bit misleading, as it also configures loads of other things 😆

Reply:
Ah, didn't know that. This requires changing the axes, though, and `seriesConfig` only applies to series, right?

Reply:
The `seriesConfig` is an object that defines aspect of the chart that is specific to the series' type. For example the way we compute axis extremums is not the same for line series or scatter series.